src/backend/binaryen/utils.ts

```diff
@@ -1817,21 +1848,56 @@ export namespace FunctionalFuncs {
                 break;
             }
             default:
-                if (rightValueType.kind === ValueTypeKind.NUMBER) {
+                if (
+                    leftValueType.kind === ValueTypeKind.NUMBER ||
+                    rightValueType.kind === ValueTypeKind.NUMBER
+                ) {
+                    const isLeftStatic =
+                        leftValueType.kind === ValueTypeKind.NUMBER
+                            ? true
+                            : false;
                     res = operateF64F64ToDyn(
                         module,
                         leftValueRef,
                         rightValueRef,
                         opKind,
-                        true,
+                        isLeftStatic,
+                        !isLeftStatic,
                     );
-                } else if (rightValueType.kind === ValueTypeKind.STRING) {
+                    if (
+                        opKind === ts.SyntaxKind.LessThanEqualsToken ||
+                        opKind === ts.SyntaxKind.LessThanToken ||
+                        opKind === ts.SyntaxKind.GreaterThanEqualsToken ||
+                        opKind === ts.SyntaxKind.GreaterThanToken
+                    ) {
```

Comment:
Take these opKinds out is weird,  I know you want to convert the output to i32 type, but in semantic tree, the `BinaryExpr`  is already `Boolean` type, so in line 1996 to line 2000 may ignore this precondition.

Maybe we can change line 1996 - line 2000 to:
```ts
if (binaryen.getExpressionType(operateNumber) === binaryen.i32) {
     return operateNumber;
} else {
    return generateDynNumber(module, operateNumber);
}
```

In this way we don't have to deal with these opKinds separately.